src/Core/Scheduler/Providers/WMS_Provider.js

```diff
@@ -69,13 +82,21 @@ WMS_Provider.prototype.preprocessDataLayer = function preprocessDataLayer(layer)
                   }&HEIGHT=${layer.width}`;
 };
 
-WMS_Provider.prototype.tileInsideLimit = function tileInsideLimit(tile, layer) {
-    return tile.level >= layer.options.zoom.min &&
-        tile.level <= layer.options.zoom.max &&
-        layer.extent.intersectsExtent(tile.extent);
+WMS_Provider.prototype.tileInsideLimit = function tileInsideLimit(tile, layer, targetLevel) {
+    for (const coord of tile.getCoordsForLayer(layer)) {
+        let c = coord;
+        // override
+        if (targetLevel < c.zoom) {
+            c = OGCWebServiceHelper.WMTS_WGS84Parent(coord, targetLevel);
+        }
+        if (c.zoom < layer.options.zoom.min || c.zoom > layer.options.zoom.max) {
+            return false;
+        }
+    }
+    return true;
 };
 
-WMS_Provider.prototype.getColorTexture = function getColorTexture(tile, layer, targetLevel) {
+WMS_Provider.prototype.getColorTexture = function getColorTexture(coordWMTS, layer, tile, targetLevel) {
```

Comment:
`coordWMTS` name seems out of place in a WMTS_Provider. Can you explain why you use this name?